frontend/public/components/modals/configure-ns-pull-secret-modal.jsx

```diff
@@ -77,6 +77,15 @@ const generateSecretData = (formData) => {
   return Base64.encode(JSON.stringify(config));
 };
 
+const generateResource = (name, namespace) => {
```

Comment:
We shouldn't be generating fake resources. It's better to add a separate `k8sPatchByName` or similar that just takes in the name and namespace.

Reply:
The reason this is a problem is that `k8sPatch` could be updated to read some other value like UID in the future, expecting it to be a real k8s resource. That would break this modal, and we wouldn't know. It's better to be explicit and create a version that takes only name and namespace as parameters.